Compilation fails for v1.9.1 in go1.16 environment

In the latest release v1.9.1, even though the mentioned version in `go.mod` file is `1.16,` the compilation fails in a go1.16 env.

Output:

```
$ go test ./...
# github.com/stretchr/testify/assert
../../go/pkg/mod/github.com/stretchr/testify@v1.8.4/assert/assertions.go:94:23: field.IsExported undefined (type reflect.StructField has no field or method IsExported)
note: module requires Go 1.20
FAIL    github.com/gomodule/redigo/redis [build failed]
FAIL    github.com/gomodule/redigo/redisx [build failed]
FAIL

$ go version
go version go1.16.15 linux/amd64
```

--- a/go.mod
+++ b/go.mod
@@ -1,8 +1,14 @@
 module github.com/gomodule/redigo
 
-go 1.16
+go 1.17
 
 require github.com/stretchr/testify v1.8.4
+
+require (
+	github.com/davecgh/go-spew v1.1.1 // indirect
+	github.com/pmezard/go-difflib v1.0.0 // indirect
+	gopkg.in/yaml.v3 v3.0.1 // indirect
+)
 
 retract (
 	v2.0.0+incompatible // Old development version not maintained or published.

